tests/migration_test.go

```diff
@@ -1827,6 +1827,114 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
 				})
 			})
 		})
+		Context("with a host-model cpu", func() {
+			getNodeHostModel := func(node *k8sv1.Node) (hostModel string) {
+				for key, _ := range node.Labels {
+					if strings.HasPrefix(key, v1.HostModelCPULabel) {
+						hostModel = strings.TrimPrefix(key, v1.HostModelCPULabel)
+						break
+					}
+				}
+				Expect(hostModel).ToNot(BeEmpty(), "must find node's host model")
+				return hostModel
+			}
+			getNodeHostRequiredFeatures := func(node *k8sv1.Node) (features []string) {
+				for key, _ := range node.Labels {
+					if strings.HasPrefix(key, v1.HostModelRequiredFeaturesLabel) {
+						features = append(features, strings.TrimPrefix(key, v1.HostModelRequiredFeaturesLabel))
+					}
+				}
+				return features
+			}
+			isModelSupportedOnNode := func(node *k8sv1.Node, model string) bool {
+				for key, _ := range node.Labels {
+					if strings.HasPrefix(key, v1.CPUModelLabel) && strings.Contains(key, model) {
+						return true
+					}
+				}
+				return false
+			}
+			expectFeatureToBeSupportedOnNode := func(node *k8sv1.Node, features []string) {
+				isFeatureSupported := func(feature string) bool {
+					for key, _ := range node.Labels {
+						if strings.HasPrefix(key, v1.CPUFeatureLabel) && strings.Contains(key, feature) {
+							return true
+						}
+					}
+					return false
+				}
+
+				supportedFeatures := make(map[string]bool)
+				for _, feature := range features {
+					supportedFeatures[feature] = isFeatureSupported(feature)
+				}
+
+				Expect(supportedFeatures).Should(Not(ContainElement(false)))
+			}
+			getOtherNodes := func(nodeList *k8sv1.NodeList, node *k8sv1.Node) (others []*k8sv1.Node) {
+				for _, curNode := range nodeList.Items {
+					if curNode.Name != node.Name {
+						others = append(others, &curNode)
+					}
+				}
+				return others
+			}
+			isHeterogeneousCluster := func() bool {
+				nodes := tests.GetAllSchedulableNodes(virtClient)
+				for _, node := range nodes.Items {
+					hostModel := getNodeHostModel(&node)
+					otherNodes := getOtherNodes(nodes, &node)
+
+					foundSupportedNode := false
+					foundUnsupportedNode := false
+					for _, otherNode := range otherNodes {
+						if isModelSupportedOnNode(otherNode, hostModel) {
+							foundSupportedNode = true
+						} else {
+							foundUnsupportedNode = true
+						}
+
+						if foundSupportedNode && foundUnsupportedNode {
+							return true
+						}
+					}
+				}
+
+				return false
+			}
+
+			It("should migrate only to nodes supporting right cpu model", func() {
+				if !isHeterogeneousCluster() {
```

Comment:
How do we achieve a heterogeneous cluster in kubevirtci?

Reply:
I think we can drop this check and mix up the success criteria a bit so it's accurate for both heterogeneous and homogeneous.